tests/countries/test_guinea.py

```diff
@@ -0,0 +1,269 @@
+#  holidays
+#  --------
+#  A fast, efficient Python library for generating country, province and state
+#  specific sets of holidays on the fly. It aims to make determining whether a
+#  specific date is a holiday as fast and flexible as possible.
+#
+#  Authors: Vacanza Team and individual contributors (see AUTHORS.md file)
+#           dr-prodigy <dr.prodigy.github@gmail.com> (c) 2017-2023
+#           ryanss <ryanssdev@icloud.com> (c) 2014-2017
+#  Website: https://github.com/vacanza/holidays
+#  License: MIT (see LICENSE file)
+
+from datetime import date
+from unittest import TestCase
+
+from holidays.countries.guinea import Guinea, GN, GIN
+from tests.common import CommonCountryTests
+
+
+class TestGuinea(CommonCountryTests, TestCase):
+    @classmethod
+    def setUpClass(cls):
+        super().setUpClass(Guinea, years=range(1959, 2050))
+        cls.no_estimated_holidays = Guinea(years=range(2015, 2025), islamic_show_estimated=False)
```

Comment:
```suggestion
        years = range(1959, 2050)
        super().setUpClass(Guinea, years=years)
        cls.no_estimated_holidays = Guinea(years=years, islamic_show_estimated=False)
```
Yes, it seems that Pakistan was a wrong example. Indonesia or the Philippines are more suitable. This is the range of years I meant by "whole".